From 66641e397de99fe6e1d8d38d792e431de0d55ab9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 14:51:32 +0200
Subject: [PATCH] [SW-0] pll: do not depend on divisor number

Currently, we assume that the divisor 0 is always the undivided
output of the PLL.
There is no reason to assume this (although it is true), so remove
this assumption and make the code more generic and readable.

Change-Id: I8b35066ebddf6b965af5e38ee32eb1ccb83bc66e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index c57ac08..68940f0 100644
--- a/pll.c
+++ b/pll.c
@@ -146,7 +146,7 @@
 		if (bypass)
 			writel(0, reg);
 		else
-			writel((sel || (cfg->div_factor[sel]))?3:1, reg);
+			writel(cfg->div_factor[sel] ? 3 : 1, reg);
 	}
 
 	/* just to be on a safe side, wait for bypass to propagate */
